cdap-unit-test/src/main/java/co/cask/cdap/test/UnitTestManager.java

```diff
@@ -156,13 +168,25 @@ public ApplicationManager deployApplication(Id.Namespace namespace, Class<? exte
       ApplicationSpecification appSpec = configurer.createSpecification();
       Location deployedJar = appFabricClient.deployApplication(namespace, appSpec.getName(),
                                                                applicationClz, appConfig, bundleEmbeddedJars);
-      return appManagerFactory.create(Id.Application.from(namespace, appSpec.getName()), deployedJar, appSpec);
+      return appManagerFactory.create(Id.Application.from(namespace, appSpec.getName()), deployedJar);
     } catch (Exception e) {
       throw Throwables.propagate(e);
     }
   }
 
   @Override
+  public ApplicationManager deployApplication(Id.Application appId,
```

Comment:
Just curious, is there any way we can reduce the code duplication between this class and IntegrationTestManager?